logic/transaction.js

```diff
@@ -65,1057 +51,1165 @@ __private.types = {};
+			amount,
+			'balance',
 			transaction,
 			sender
-		) || false;
-	if (!fee || transaction.fee !== fee) {
-		return setImmediate(cb, 'Invalid transaction fee');
-	}
-
-	// Check amount
-	if (
-		transaction.amount < 0 ||
-		transaction.amount > constants.totalAmount ||
-		String(transaction.amount).indexOf('.') >= 0 ||
-		transaction.amount.toString().indexOf('e') >= 0
-	) {
-		return setImmediate(cb, 'Invalid transaction amount');
-	}
+		);
 
-	// Check confirmed sender balance
-	var amount = new bignum(transaction.amount.toString()).plus(
-		transaction.fee.toString()
-	);
-	var senderBalance = this.checkBalance(amount, 'balance', transaction, sender);
+		if (senderBalance.exceeded) {
+			return setImmediate(cb, senderBalance.error);
+		}
 
-	if (senderBalance.exceeded) {
-		return setImmediate(cb, senderBalance.error);
-	}
+		// Check timestamp
+		if (slots.getSlotNumber(transaction.timestamp) > slots.getSlotNumber()) {
+			return setImmediate(
+				cb,
+				'Invalid transaction timestamp. Timestamp is in the future'
+			);
+		}
 
-	// Check timestamp
-	if (slots.getSlotNumber(transaction.timestamp) > slots.getSlotNumber()) {
-		return setImmediate(
-			cb,
-			'Invalid transaction timestamp. Timestamp is in the future'
+		// Call verify on transaction type
+		__private.types[transaction.type].verify.call(
+			this,
+			transaction,
+			sender,
+			err => {
+				if (err) {
+					return setImmediate(cb, err);
+				}
+				// Check for already confirmed transaction
+				return this.checkConfirmed(transaction, cb);
+			},
+			tx
 		);
 	}
 
-	// Call verify on transaction type
-	__private.types[transaction.type].verify.call(
-		this,
-		transaction,
-		sender,
-		err => {
-			if (err) {
-				return setImmediate(cb, err);
-			}
-			// Check for already confirmed transaction
-			return self.checkConfirmed(transaction, cb);
-		},
-		tx
-	);
-};
+	/**
+	 * Verifies signature for valid transaction type.
+	 *
+	 * @param {transaction} transaction
+	 * @param {publicKey} publicKey
+	 * @param {signature} signature
+	 * @throws {Error}
+	 * @returns {boolean}
+	 * @todo Add description for the params
+	 */
+	verifySignature(transaction, publicKey, signature) {
+		if (!__private.types[transaction.type]) {
+			throw `Unknown transaction type ${transaction.type}`;
+		}
 
-/**
- * Verifies signature for valid transaction type.
- *
- * @param {transaction} transaction
- * @param {publicKey} publicKey
- * @param {signature} signature
- * @throws {Error}
- * @returns {boolean}
- * @todo Add description for the params
- */
-Transaction.prototype.verifySignature = function(
-	transaction,
-	publicKey,
-	signature
-) {
-	if (!__private.types[transaction.type]) {
-		throw `Unknown transaction type ${transaction.type}`;
-	}
+		if (!signature) {
+			return false;
+		}
 
-	if (!signature) {
-		return false;
-	}
+		let res;
 
-	var res;
+		try {
+			const bytes = this.getBytes(transaction, true, true);
+			res = this.verifyBytes(bytes, publicKey, signature);
```

Comment:
`const res =`, and then return it here
or just `return this.verifyBytes(bytes, publicKey, signature);`